### PR TITLE
JavaRDD extension functions + iterators

### PR DESCRIPTION
**Pull request by Jolanrensen**
https://github.com/Kotlin/kotlin-spark-api/issues/151, Work in progress...

### CODE REVIEW

**Comment by Jolanrensen** [comment]
just needs more tests

File: kotlin-spark-api/src/main/kotlin/org/jetbrains/kotlinx/spark/api/Conversions.kt [hunk 1]
```diff
@@ -45,6 +45,7 @@ import scala.collection.mutable.Seq as ScalaMutableSeq
 import scala.collection.mutable.Set as ScalaMutableSet
 import org.apache.spark.streaming.State
 
+
```
Comment (asm0dey) [inline kotlin-spark-api/src/main/kotlin/org/jetbrains/kotlinx/spark/api/Conversions.kt:48]: This line is definitely obsolete

File: kotlin-spark-api/src/main/kotlin/org/jetbrains/kotlinx/spark/api/RddDouble.kt [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+fun JavaRDD<Double>.sum(): Double = toJavaDoubleRDD().sum()
+
+/**
+ * Return a [org.apache.spark.util.StatCounter] object that captures the mean, variance and
+ * count of the RDD's elements in one operation.
+ */
+fun JavaRDD<Double>.stats(): StatCounter = toJavaDoubleRDD().stats()
+
+/** Compute the mean of this RDD's elements. */
+fun JavaRDD<Double>.mean(): Double = toJavaDoubleRDD().mean()
```
Comment (asm0dey) [inline kotlin-spark-api/src/main/kotlin/org/jetbrains/kotlinx/spark/api/RddDouble.kt:29]: Is there mean/average for other types than Double?

**Review by asm0dey — COMMENTED** [review]
Couple o small comments, but generally LGTM

File: kotlin-spark-api/src/main/kotlin/org/jetbrains/kotlinx/spark/api/RddDouble.kt (hunk 2 above)
Comment (Jolanrensen) [reply kotlin-spark-api/src/main/kotlin/org/jetbrains/kotlinx/spark/api/RddDouble.kt:29]: Nope, although there is the implicit number conversion in scala. In Java there's just `JavaDoubleRDD` and in Scala `DoubleRDDFunctions` for `RDD[Double]`, but it appears I can call the same functions with an `RDD[Int]`... I'll see if I can add conversions for any `Number`!

Comment (Jolanrensen) [reply kotlin-spark-api/src/main/kotlin/org/jetbrains/kotlinx/spark/api/RddDouble.kt:29]: good one actually!

**Pull request merged**